src/python/dxpy/scripts/dx.py

```diff
@@ -3787,6 +3787,117 @@ def publish(args):
     except:
         err_exit()
 
+def archive(args):
+    dx_archive_errors = [InvalidState, ResourceNotFound, PermissionDenied]
+
+    # resolve paths
+    target_project = None    
+    target_folder = None
+    target_files = []
+    for path in args.path:
+        try: 
+            project_id, folderpath, file_results = resolve_existing_path(path, expected_classes=None, allow_mult=True, all_mult=args.all)
```

Comment:
If there are many calls to this this will certainly be expensive.  We may want to avoid calling this procedure at all and be a bit more conservative about the requirements of what is in `args.path`.  For example, we may require that this list contains either valid folder paths--as verified by a single call made at the beginning to `project-xxxx/describe` or a string of the form `project-xxxx:file-yyyy` which can be validated by a single bulk call, e.g. to `findDataObjects` or something like that.  This approach would minimize API calls made to the platform by performing bulk operations when necessary and minimize load to the platform.  Another approach would be to separate out the arguments into two groups: folders and files.  Files must always be accompanied with a 'project' to help us minimize load.  For this feature in particular I think it's important to think about the scale of the inputs because my feeling is that users tend to like to archive lots of files at a time.  It's natural to do that by folder or project, but I can see users providing lots of files in some cases and IMHO and we should either (1) address this in the design as a limit we impose and error out on the client side or (2) protect ourselves via the method described above.

Reply:
For now it allows either one folder path or array of files in one project, so I guess it would not be so expensive? 
If it is a folder path it is passed directly to the server after being resolved without extra API call; and if we have files, resolving the path calls 'describe' on each (But hopefully the user would paste 1000 files here :D ) 

Also using`resolve_existing_path` here could help if there is more than one file with the same name. In that case the user can choose whether to pick some of them.

Reply:
If user supplies paths as `project-xxxx:file-yyyy` or `file-yyyy` (in which case project must be set), can we skip resolution entirely and pass paths as specified by the user to project-xxxx/archive?   We can also recommend this input format for bulk archiving calls.

Reply:
I checked the nucleus code and with the given project id
- if 'files', it will check if all inputs are valid IDs and if all are in that project
- if 'folder', it will check if the folder is in that project

My guess is we can simply prepare the inputs into 1 project-xxxx, and a list of file-xxxx or one clean/folder/path. However if proj-xxxx:/path/to/file is given we still need to resolve it.